examples/platform/telink/common/src/AppTaskCommon.cpp

```diff
@@ -108,6 +109,22 @@ Identify sIdentify = {
 uint8_t sTestEventTriggerEnableKey[TestEventTriggerDelegate::kEnableKeyLength] = { 0x00, 0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77,
                                                                                    0x88, 0x99, 0xaa, 0xbb, 0xcc, 0xdd, 0xee, 0xff };
 #endif
+
+class AppCallbacks : public AppDelegate
+{
+    bool isComissioningStarted;
+
+public:
+    void OnCommissioningSessionStarted() override { isComissioningStarted = true; }
+    void OnCommissioningSessionStopped() override { isComissioningStarted = false; }
```

Comment:
OnCommissioningSessionStopped() does not actually mean "PASE session was torn down".  In fact, OnCommissioningSessionStarted() will never be followed by OnCommissioningSessionStopped() as far as I can tell.  AppDelegate _really_ needs better documentation.

See also https://github.com/project-chip/connectedhomeip/pull/25215 and the discussion in that PR.

Reply:
Again, if OnCommissioningSessionStarted is called you will _not_ get a call to OnCommissioningSessionStopped.  So why do you implement OnCommissioningSessionStopped at all?

Reply:
@bzbarsky-apple 

We have the case, when commissioning has started and has finished successfully. In this case the OnCommissioningSessionStopped() will not be called at all. You are right. But we don't need this method to be called if the commissioning has passed. 

In case if commissioning session was terminated and was not passed (failed), the _CommissioningWindowManager::HandleFailedAttempt_ is called. And the _mAppDelegate->OnCommissioningSessionStopped();_ is called from this method. 

So we have the case, when the commissioning has started. The OnCommissioningSessionStarted() is setting the flag to prevent the BLE to be switched off. We are able to pass the commissioning procedure and switched to Thread network. 
The BLE is switched off be BLE Manager, because Telink MCU does not support concurrent mode. In this case the commissioning timer does not need at all, and we don't need to take care about it status and the flag status.

If the commissioning has started and had a failure by any reason, the _CommissioningWindowManager::HandleFailedAttempt_ has called and called the _mAppDelegate->OnCommissioningSessionStopped()_ , which is resetting the flag, set up by _OnCommissioningSessionStarted()_ , so the commissioning window timer is reenabled again, and BLE will be switched off in case if no connection attempt.